src/datatypes/src/vectors/struct_vector.rs

```diff
@@ -323,6 +323,96 @@ impl StructVectorBuilder {
         }
         self.null_buffer.append_null();
     }
+
+    /// Try to merge (and consume the data of) other struct vector builder into this one.
+    pub(crate) fn try_merge(&mut self, other: &mut StructVectorBuilder) -> Result<()> {
+        if let Some(other) = other.null_buffer.finish_cloned() {
+            self.null_buffer.append_buffer(&other)
+        } else {
+            self.null_buffer.append_n_non_nulls(other.len());
+        }
+
+        for (i, field) in self.fields.fields().iter().enumerate() {
+            let builder = &mut self.value_builders[i];
+
+            if let Some(j) = other
+                .fields
+                .fields()
+                .iter()
+                .position(|other| other.name() == field.name())
```

Comment:
is this nested for loop in a hot path? I wonder if we need to optimize this, for example, keep fields sorted by default